crates/nu-command/src/filters/take/take_.rs

```diff
@@ -68,48 +80,92 @@ impl Command for Take {
 
         match input {
             PipelineData::Value(val, _) => match val {
-                Value::List { vals, .. } => Ok(vals
-                    .into_iter()
-                    .take(rows_desired)
-                    .into_pipeline_data(ctrlc)
-                    .set_metadata(metadata)),
+                Value::List { vals, .. } => {
+                    if take_last {
+                        Ok(vals
+                            .into_iter()
+                            .rev()
+                            .take(rows_desired)
```

Comment:
This uses excessive memory and should be replaced by a `VecDeque` ring buffer as suggested in #7178

Reply:
Good catch! Will do.